protocol.cpp

```diff
@@ -253,6 +261,21 @@ int redis_protocol::authenticate(const char *credentials)
     return size;
 }
 
+int redis_protocol::configure_protocol(enum PROTOCOL_CONFIGURATION conf) {
+    int size = 0;
+    if (conf == PROTOCOL_CONF_RESP2 || conf == PROTOCOL_CONF_RESP3) {
+        m_resp3 = conf == PROTOCOL_CONF_RESP3;
+        size = evbuffer_add_printf(m_write_buf,
```

Comment:
Do we want to maintain a backwards compatible mode where we don't send `HELLO` and we assume we're using RESP2? Without it, we'd lose compatibility with Redis <6.0.

Reply:
Not sure I got it. In the current implementation if the user doesn't add `--resp` he will get the old behavior (will not get into this piece of code).